test/spec/app_spec.rb

```diff
@@ -585,28 +580,34 @@
 
     it "should add and remove source languages", bin6: true  do
       api_create_team_project_and_source_and_redirect_to_source('GOT', 'https://twitter.com/GameOfThrones')
-      sleep 5
-      element = @driver.find_element(:class, "source__edit-button")
+      wait_for_selector("source__tab-button-account",:class)
+      element = wait_for_selector("source__edit-button",:class)
       element.click
-      sleep 1
-      @driver.find_element(:class, "source__edit-addinfo-button").click
-      sleep 1
-      @driver.find_element(:class, "source__add-languages").click
+      wait_for_selector("source__edit-buttons-cancel-save",:class)
+      element = wait_for_selector("source__edit-addinfo-button",:class)
+      element.click
+      sleep 2
+      element = wait_for_selector("source__add-languages",:class)
+      element.click
+      sleep 2
       fill_field("sourceLanguageInput", "Acoli", :id)
       @driver.action.send_keys(:down).perform
       @driver.action.send_keys(:return).perform
-      sleep 2
+      sleep 1
+      wait_for_size_change(0, "sourceLanguageInput",:id)
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source-tags__tag",:class)  
       expect(@driver.page_source.include?('Acoli')).to be(true)
-      sleep 1 until element = @driver.find_element(:class, "source__edit-button")
+      element = wait_for_selector("source__edit-button",:class)
       element.click
       sleep 1
-      list = @driver.find_elements(:css => "div.source-tags__tag svg")
-      list[0].click
+      elements =wait_for_selector_list("div.source-tags__tag svg")
+      elements[0].click
       sleep 1
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source__tab-button-media",:class)
       expect(@driver.page_source.include?('Acoli')).to be(false)
```

Comment:
Same as above... shouldn't we copy `expect(@driver.page_source.include?('Acoli')).to be(false)` to after line 607 (before `@driver.navigate.refresh`) in order to be sure that the language was removed without needing to refresh the page?

Reply:
The idea is to see if the tag with the language was deleted in the page not in the edit mode.

Reply:
But after we press "Enter" we exit edit mode. I mean that we don't need to refresh the page to see that the item was actually saved. We can just press "Save". Here: https://www.youtube.com/watch?v=a7w4qDacORA

Reply:
Replacing the refresh by press save